agent/battle/vision/parser.py

```
@@ -0,0 +1,125 @@
+        evidence_id=evidence_id or str(data.get("evidence_id", "")),
+    )
+
+def _load(raw: str) -> dict[str, Any]:
+    text = (raw or "").strip()
+    match = re.search(r"```(?:json)?\s*(\{.*?\})\s*```", text, re.DOTALL | re.IGNORECASE)
+    if match:
+        text = match.group(1)
+    try:
+        value = json.loads(text)
+    except json.JSONDecodeError as exc:
+        raise VisionParseError(f"invalid JSON: {exc.msg}") from exc
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)